webapp/src/ts/reducers/tasks.ts

```diff
@@ -14,20 +14,60 @@ const initialState = {
   },
 };
 
-const orderByDueDate = (t1, t2) => {
-  const lhs = t1?.dueDate;
-  const rhs = t2?.dueDate;
-  if (!lhs && !rhs) {
-    return 0;
-  }
-  if (!lhs) {
-    return 1;
+/**
+ * Task prioritization algorithm that combines:
+ * 1. Overdue status (most urgent)
+ * 2. Due today status (high urgency)
+ * 3. Priority (importance)
+ * 4. Due date (soonest first)
+ * 5. Tasks without due dates (lowest priority)
+ * 
+ * Sorting rules (in order):
+ * 1. Overdue tasks appear first (most urgent), sorted by priority (higher first)
+ * 2. Tasks due today appear next, sorted by priority (higher first)
+ * 3. Then sort other dates too by date and priority (high to low)
+ * 4. For equal priority, sort by due date (earlier first)
+ * 5. Tasks without due dates appear last
+ */
+const orderByDueDateAndPriority = (t1, t2) => {
+  const getPriorityScore = (t) => {
+    if (t?.priority === 'high') {
+      return 10;
+    } else if (t?.priority === 'medium') {
+      return 6;
+    } else if (typeof t?.priority === 'string') {
+      return 0;
+    } else {
+      return t?.priority ?? 0;
+    }
+  };
```

Comment:
I think we should not hardcode **any values**. 
If the priority is a string, we always consider it the lowest priority (for example, have this function return NaN, and check for NaN when comparing priorities.